### PR TITLE
A stab at better pseudocode. Not sure if this is helpful or not.

### PR DESCRIPTION
**Pull request by danmarg**

### CODE REVIEW

File: mta-sts.txt [hunk 1]
```diff
@@ -618,24 +618,61 @@ Margolis, et al.         Expires January 9, 2017               [Page 11]
...
-      if invalid_mx_or_tls(policy):
-        reject_message()
-        update_cache = false
-  if update_cache:
-    cache(policy)
+7.  Appendix 1: Delivery Pseudocode
+
+   Assume a persistent per-recipient-domain cache of an ordered list of
+
+                    {
```
Comment (eyeofthenico) [inline mta-sts.txt]: Reason for the excessive indentation?

File: mta-sts.txt [hunk 2]
```diff
@@ -618,24 +618,61 @@ Margolis, et al.         Expires January 9, 2017               [Page 11]
...
+                      STSPolicy policy
+                      boolean was_successfully_applied
+                      string version_id
+                    }
+
+   Delivery to a given domain looks like
+
+function attempt_policy(policy) {
+  for each MX in MX candidates {
```
Comment (eyeofthenico) [inline mta-sts.txt]: Should we keep mention of MX ordering here?

File: mta-sts.txt [hunk 3]
```diff
@@ -618,24 +618,61 @@ Margolis, et al.         Expires January 9, 2017               [Page 11]
...
-      policy = policy_from_https_endpoint()  // fetch and authenticate #2!
-      update_cache = true
-      if invalid_mx_or_tls(policy):
-        reject_message()
-        update_cache = false
-  if update_cache:
-    cache(policy)
+7.  Appendix 1: Delivery Pseudocode
+
+   Assume a persistent per-recipient-domain cache of an ordered list of
```
Comment (eyeofthenico) [inline mta-sts.txt]: Why "Ordered list"?

**Comment by eyeofthenico** [comment]
I am a bit confused by the logic to follow here.

**Comment by danmarg** [comment]
Re: ordered list

Ordering it makes it easier to do the "insert a new (possibly not yet successfully applied) policy but be able to fall back to the old policy". This is to accommodate this: 

>    Finally, an STS Policy MUST NOT be be used to reject mail until it
>    has been successfully validated when delivering at least one message
>    to the Policy Domain.  This is to limit the risk of misconfigurations
>    when deploying new policies.

Re: MX candidates, etc: 

Yeah, could be fleshed out better, as well. :-/

**Pull request closed without merging**